hal-sim/hal_impl/functions.py

```diff
@@ -1850,5 +1863,14 @@ def clearAllPCMStickyFaults(module, status):
     status.value = 0
     return False
 
+def setOneShotDuration(solenoidPortHandle, durMS, status):
+    status.value = 0
+    hal_data['pcm'][solenoidPortHandle.module][solenoidPortHandle.pin]['one_shot_duration'] = durMS
+
+def fireOneShot(solenoidPortHandle, status):
+    status.value = 0
+    # TODO: need to schedule a callback to implement this somehow
```

Comment:
I don't really have a good idea on how to make this work. We would need some hal mechanism to schedule callbacks after some specified time.